### PR TITLE
Added /skytils help + Show terminals you have done

### PR DESCRIPTION
**Pull request by AzuredBlue**
im proud of myself

### CODE REVIEW

**Review by My-Name-Is-Jeff — APPROVED** [review]

**Review by My-Name-Is-Jeff — APPROVED** [review]

File: src/main/java/skytils/skytilsmod/commands/SkytilsCommand.java [hunk 1]
```diff
@@ -106,6 +106,11 @@ public void processCommand(ICommandSender sender, String[] args) throws CommandE
...
+                if (args.length == 1) {
+                    player.addChatMessage(new ChatComponentText("\n" + EnumChatFormatting.AQUA + "Skytils" + " Version " + Skytils.VERSION + "\n" + EnumChatFormatting.GOLD + " /skytils" + EnumChatFormatting.AQUA + " - Opens the main GUI" + "\n" + EnumChatFormatting.GOLD + " /skytils help" + EnumChatFormatting.AQUA + " - Returns this message" + "\n" + EnumChatFormatting.GOLD + " /skytils setkey <key>" + EnumChatFormatting.AQUA + " - Sets your API key" + "\n" + EnumChatFormatting.GOLD + " /skytils griffin refresh" + EnumChatFormatting.AQUA + " - Forces a refresh for the Griffin Burrow waypoints" + "\n" + EnumChatFormatting.GOLD + " /skytils reload <aliases/data>" + EnumChatFormatting.AQUA + " - Forces Skytils to re-fetch your command aliases or solutions from the data repository." + "\n" + EnumChatFormatting.GOLD + "/reparty" + EnumChatFormatting.AQUA + " - Disbands and sends a party invite to everyone who was in your party." + "\n" + EnumChatFormatting.GOLD + "/armorcolor <set/clear/clearall>" + EnumChatFormatting.AQUA + " - Changes the color of an armor piece to the hexcode or decimal color provided. (Alias is /armourcolour)"));
```
Comment (Sychic) [inline src/main/java/skytils/skytilsmod/commands/SkytilsCommand.java:111]: You should maybe split this up into multiple lines just to make it more readable

File: src/main/java/skytils/skytilsmod/features/impl/dungeons/DungeonsFeatures.java [hunk 2]
```diff
@@ -38,12 +37,28 @@
...
+                event.setCanceled(true);
+            }
+        }
+        if (Skytils.config.hideBlessings && Utils.inDungeons) {
+            if (unformatted.startsWith("DUNGEON BUFF!") || unformatted.startsWith("A Blessing") || unformatted.contains("has obtained Blessing of") || unformatted.contains("Grants you") || unformatted.contains("Granted you") || unformatted.contains("found a Wither Essence!")) {
+                event.setCanceled(true);
+            }
+        }
+        if (Skytils.config.hideMilestones && Utils.inDungeons) {
+            if (unformatted.startsWith("Mage Milestone") || unformatted.startsWith("Berserk Milestone") || unformatted.startsWith("Archer Milestone") || unformatted.startsWith("Tank Milestone") || unformatted.startsWith("Healer Milestone")) {
```
Comment (Sychic) [inline src/main/java/skytils/skytilsmod/features/impl/dungeons/DungeonsFeatures.java:58]: Why not change this to just a `.contains("Milestone")`

Comment (My-Name-Is-Jeff) [reply src/main/java/skytils/skytilsmod/features/impl/dungeons/DungeonsFeatures.java:58]: What if someone in the party says "I need Milestones"
If you went this route you would need regex

File: src/main/java/skytils/skytilsmod/core/GuiManager.java [hunk 3]
```diff
@@ -15,9 +15,9 @@
 import java.util.Map;
 
 public class GuiManager {
-    private Map<Integer, GuiElement> elements = new HashMap<>();
+    private final Map<Integer, GuiElement> elements = new HashMap<>();
     private int counter = 0;
-    private Map<String, GuiElement> names = new HashMap<>();
+    private final Map<String, GuiElement> names = new HashMap<>();
```
Comment (Sychic) [inline src/main/java/skytils/skytilsmod/core/GuiManager.java:20]: what's the point of these finals?

File: src/main/java/skytils/skytilsmod/core/Config.java [hunk 4]
```diff
@@ -556,6 +556,42 @@
...
+            description = "Hides damage milestones while in dungeons",
+            category = "Dungeons",
+            subcategory = "Quality of Life"
+    )
+    public boolean hideMilestones = false;
+
+    @Property(
+            type = PropertyType.SWITCH,
+            name = "Hide Farming RNG Titles",
+            description = "Removes the titles that show up after getting an RNG with Pumpkin Dicer / Melon Dicer",
```
Comment (Sychic) [inline src/main/java/skytils/skytilsmod/core/Config.java:589]:
```suggestion
            description = "Removes the titles that show up after getting a drop with Pumpkin Dicer / Melon Dicer",
```

File: src/main/java/skytils/skytilsmod/core/GuiManager.java (hunk 3 above)
Comment (AzuredBlue) [reply src/main/java/skytils/skytilsmod/core/GuiManager.java:20]: didnt even write those ¯\_(ツ)_/¯ probably lily's commit that didnt get updated

File: src/main/java/skytils/skytilsmod/features/impl/dungeons/DungeonsFeatures.java [hunk 5]
```diff
@@ -38,12 +37,28 @@
...
     public void onChat(ClientChatReceivedEvent event) {
         if (!Utils.inSkyblock) return;
         String unformatted = StringUtils.stripControlCodes(event.message.getUnformattedText());
 
+
+        if (Skytils.config.hideAbilities && Utils.inDungeons) {
+            if (unformatted.contains("is now available!") || unformatted.contains("is ready to use!")) {
+                event.setCanceled(true);
+            }
+        }
```
Comment (Sychic) [inline src/main/java/skytils/skytilsmod/features/impl/dungeons/DungeonsFeatures.java:52]: You can maybe consider making this also go into the spam gui element

File: src/main/java/skytils/skytilsmod/features/impl/misc/MiscFeatures.java [hunk 6]
```diff
@@ -29,6 +31,19 @@ public void onBossBarSet(BossBarEvent.Set event) {
...
+        if (event.packet instanceof S45PacketTitle) {
+            S45PacketTitle packet = (S45PacketTitle) event.packet;
+            if (packet.getMessage() != null) {
+                String unformatted = StringUtils.stripControlCodes(packet.getMessage().getUnformattedText());
+                if (Skytils.config.hideFarmingRNGTitles && Utils.inSkyblock && unformatted.contains("DROP!")) {
+                    event.setCanceled(true);
+                }
+            }
+        }
+    }
```
Comment (Sychic) [inline src/main/java/skytils/skytilsmod/features/impl/misc/MiscFeatures.java:45]: Maybe add a way to turn off the noises too

**Review by Sychic — APPROVED** [review]
merge remote into local

**Comment by Sychic** [comment]
Why did you remove the stuff in your commit and then add it back

**Review by My-Name-Is-Jeff — APPROVED** [review]

**Pull request merged**